src/components/table/column-picker/column-picker.component.ts

```diff
@@ -41,39 +70,163 @@ export class ColumnPickerComponent {
     /** Define a function that return a column move announcement */
     @Input() columnMovedAnnouncement: (column: string, delta: number) => string = this.getColumnMovedAnnouncement;
 
+    /** group settings for customising display */
+    @Input() groupSettings: ColumnPickerGroupSetting[] = [];
+
     /** Emits when the selected items change */
-    @Output() selectedChange = new EventEmitter<ReadonlyArray<string>>();
+    @Output() selectedChange = new EventEmitter<ReadonlyArray<ColumnPickerValue>>();
 
     /** Emits when the deselected items change */
-    @Output() deselectedChange = new EventEmitter<ReadonlyArray<string>>();
+    @Output() deselectedChange = new EventEmitter<ReadonlyArray<ColumnPickerValue>>();
+
+    /** The Nested tree control used for the deselect tree */
+    treeControl: NestedTreeControl<ColumnPickerTreeNode> = new NestedTreeControl<ColumnPickerTreeNode>(item => item.children);
```

Comment:
I tried the FlatTreeControl instead of nested first but there were a number of reservations I had to using it. The structure has to be laid out exactly as it's displayed - each node has a level and each node must also have a parent node. There is too much room for human error inputing the values. With the NestedTreeControl I can build the structure but still retain the original flat structure of elements and have control over how it is built.